### PR TITLE
DeadlineQos TestCase [6425]

### PR DESCRIPTION
**Pull request by laura-eprosima**
New Test Cases for LivelinessQos, DeadlineQos and LifespanQos

### CODE REVIEW

**Comment by raquelalvarezbanos** [comment]
I know it is unrelated, but could you also, as part of this PR, update the `config.py` to reflect the new release of shapes demo, v1.9.0?

File: docs/lifespan.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
+Lifespan
+==============================
+The Lifespan QoS establish the maximum validity period of the samples saved on the entity history. Each of those
+samples has a lifespan timer associated so that once it expires the linked data is removed from the entity history.
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]: Again, maybe too technical (also not strictly speaking correct, as there is one timer shared among samples, not one for each sample). You could just say something like:

> The lifespan QoS establishes the maximum validity period of the samples saved on an entity's history. When the lifespan period elapses, the corresponding sample is automatically removed.

If you want, you could also add to the previous sentence:

> Unlike the Deadline Qos, there is no callback to inform the user when this happens

(the latter is optional, just add it if you want).

File: docs/lifespan.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Before starting with the example, we need to advise you that it is complicated to understand.
+
+In this test, we are going to create two Publishers and two Subscribers, one using lifespan and the other
+without it, so you can see graphically the difference between both cases.
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]: I liked that you warned above. However, I would say it in a different way (you can even use the sentence above which I said was optional):

> Unlike other QoS, such as Deadline or Liveliness, this Qos does not provide a means to inform the user that the sample is being removed, which makes this test more complicated to illustrate. For this reason, we are going to create two publishers and two subscribers, and make only one of them use Lifespan, so that the effect of using this QoS can be graphically seen.

File: docs/lifespan.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+**Step-by-Step**
+
+First, you have to launch an instance and create the two Publishers:
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
First, launch an instance and create the two publishers:
```

File: docs/lifespan.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    - Click on Options.
+    - Select Preferences.
+    - Set the update interval to 1000.
+
+.. image:: test10_1.png
+   :scale: 60 %
+   :alt: Initial state
+   :align: center
+
+Now, create two Subscribers:
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
Now, create two subscribers:
```

File: docs/lifespan.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+5 - Create a triangle subscriber:
+    - Start eProsima Shapes-Demo. (We will refer to this instance as Instance3)
+    - Click on Subscribe.
+    - Select TRIANGLE option for Shape.
+    - Select TRANSIENT_LOCAL option for the Durability.
+    - Make sure that RELIABLE checkbox is marked.
+    - Set History to 100.
+    - Set Lifespan Duration to 50.
+
+Note that when a new subscriber matched with the publisher, due to the TRANSIENT_LOCAL durability, all the
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
Note that when a new subscriber matches with the publisher, due to the TRANSIENT_LOCAL durability, all the
```

File: docs/lifespan.rst [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+samples stored on the publisher history are sent automatically to the new subscriber.
+
+Knowing that we are going to explain what is happening on Instance2 and Instance3. As you can see, the square
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
Now let's explain what is happening on Instance2 and Instance3. As you can see, the square
```

File: docs/lifespan.rst [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+subscriber history is fulled rapidly, while the triangle subscriber one increases at the same speed as the
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
subscriber history is filled rapidly, while the triangle subscriber one increases at the same speed as the
```

File: docs/lifespan.rst [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+orange triangle publisher sends the new samples. That's why the samples of the orange triangle publisher history
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
orange triangle publisher sends the new samples. This is because samples of the orange triangle publisher history
```

File: docs/lifespan.rst [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+are removed faster than he sends new ones, due to the lifespan, while the red square publisher history samples
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
are removed faster than new ones are being sent due to the lifespan QoS, while the red square publisher history samples
```

File: docs/liveliness.rst [hunk 10]
```diff
@@ -12,13 +12,10 @@ duration as long as the DomainParticipant keeps running and remote participants
...
 The remaining kinds (MANUAL_BY_PARTICIPANT and MANUAL_BY_TOPIC) need periodically an assertion to consider the remote 
-participants as alive. That assertion can be done by writing new data or using the function *assert_liveliness*. The 
-main difference between these two kinds is that MANUAL_BY_PARTICIPANT requires only the assertion of one of its 
-entities to consider that all of them are alive, while MANUAL_BY_TOPIC requires at least one assertion per publisher 
-to consider it as alive.
+participants as alive.
 
-In this test, we are going to create a Publisher using MANUAL_BY_TOPIC and a Subscriber using MANUAL_BY_PARTICIPANT 
-and a lease duration value lower than the write rate of the publisher.
+In this test, we are going to create a Publisher and Subscriber using AUTOMATIC liveliness and a lease duration value
```
Comment (raquelalvarezbanos) [inline docs/liveliness.rst]:
```suggestion
In this test, we are going to create a publisher and subscriber using AUTOMATIC liveliness and a lease duration value
```

File: docs/liveliness.rst [hunk 11]
```diff
@@ -12,13 +12,10 @@ duration as long as the DomainParticipant keeps running and remote participants
 entities within the DomainParticipant alive.
 
 The remaining kinds (MANUAL_BY_PARTICIPANT and MANUAL_BY_TOPIC) need periodically an assertion to consider the remote 
-participants as alive. That assertion can be done by writing new data or using the function *assert_liveliness*. The 
-main difference between these two kinds is that MANUAL_BY_PARTICIPANT requires only the assertion of one of its 
-entities to consider that all of them are alive, while MANUAL_BY_TOPIC requires at least one assertion per publisher 
-to consider it as alive.
```
Comment (raquelalvarezbanos) [inline docs/liveliness.rst]: Why did you remove this explanation? Did you consider it difficult to understand?

File: docs/liveliness.rst [hunk 12]
```diff
@@ -28,32 +25,27 @@ First, you have to launch two instances and create a Publisher and a Subscriber:
...
-As you can see both matches even if they have different liveliness kinds since the standard established that they 
-are considered as compatible as long as the offered kind is greater than the required one (considering this order 
-AUTOMATIC < MANUAL_BY_PARTICIPANT < MANUAL_BY_TOPIC).
-
-If you go to the *Output Tab* on Instance2, you can observe that the subscriber is continuously losing and recovering 
-the liveliness due to the value established for the lease duration, which is lower than the publishing rate. This fact 
-makes that the liveliness timer ends up before a new sample comes from the publisher and therefore the subscriber 
-shows the liveliness lost message. After that, a new sample arrives at the subscriber side, who notifies about the 
-liveliness recover.
+Now, you have to open the Task Manager and kill the process corresponding with the publisher (Instnace1). As you can
```
Comment (raquelalvarezbanos) [inline docs/liveliness.rst]:
```suggestion
Now, open the Task Manager and kill the process corresponding to the publisher (Instnace1). As you can
```

File: docs/liveliness.rst [hunk 13]
```diff
@@ -28,32 +25,27 @@ First, you have to launch two instances and create a Publisher and a Subscriber:
...
+see, the subscriber lost the liveliness as the publisher doesn't terminate cleanly.
```
Comment (raquelalvarezbanos) [inline docs/liveliness.rst]:
```suggestion
see, the subscriber reported that liveliness was lost, as the publisher did not terminate cleanly.
```

File: docs/liveliness.rst (hunk 11 above)
Comment (laura-eprosima) [reply docs/liveliness.rst]: I thought that explanation was too technical and since the test did not use any of those kinds, I decided to remove it. But if you think it can be understood correctly, I can add it again.

File: docs/index.rst [hunk 14]
```diff
@@ -30,6 +30,8 @@ implementations of the RTPS protocol.
 	partition
 	redundancey_fault_tolerance
         liveliness
+        deadline
+        lifespan
```
[the remark this answers is not in the document]
Comment (laura-eprosima) [reply docs/index.rst]: I have reviewed it and I see it well

[the remark this answers is not in the document]
Comment (raquelalvarezbanos) [reply docs/index.rst]: OK, I see it OK now too, for some reason

File: docs/deadline.rst [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+   - Set a value for the Deadline Duration higher or equal to the one stated for the Publisher.
+     (If the value of subscriber deadline duration is lower the entities don't match)
+
+You can see know the normal functionality of the system, as the publisher is sending new samples and the subscriber
+receive them before the deadline timer expires.
+
+.. image:: test9_1.png
+   :scale: 60 %
+   :alt: State 1
+   :align: center
```
[the remark this answers is not in the document]
Comment (raquelalvarezbanos) [reply docs/deadline.rst]: So much better, thank you!

File: docs/liveliness.rst (hunk 11 above)
Comment (raquelalvarezbanos) [reply docs/liveliness.rst]: I think removing it is fine, I was just asking :)

File: docs/lifespan.rst [hunk 16]
```diff
@@ -0,0 +1,76 @@
+Lifespan
+==============================
+The Lifespan QoS establishes the maximum validity period of the samples saved on an entity's history. When
+the lifespan period elapses, the corresponding sample is automatically removed.
+
+Unlike other QoS, such as Deadline or Liveliness, this Qos does not provide a means to inform the user that
+the sample is being removed, which makes this test more complicated to illustrate. For this reason, we are
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
the sample is being removed, which makes this QoS more complicated to illustrate. For this reason, we are
```

File: docs/lifespan.rst [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+    - Select TRANSIENT_LOCAL option for the Durability.
+    - Make sure that RELIABLE checkbox is marked.
+    - Set History to 100.
+    - Set Lifespan Duration to 50.
+
+Note that when a new subscriber matches with the publisher, due to the TRANSIENT_LOCAL durability, all the
+samples stored on the publisher history are sent automatically to the new subscriber.
+
+Now let's explain what is happening on Instance2 and Instance3. As you can see, the square
+subscriber history is filled rapidly, while the triangle subscriber one increases at the same speed as the
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
subscriber history is filled rapidly, while the triangle subscriber one is filled at the same speed as the
```

File: docs/lifespan.rst [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+orange triangle publisher sends the new samples. This is because samples of the orange triangle publisher history
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
orange triangle publisher sends new samples. This is because in the second case, samples in the orange triangle publisher history
```

File: docs/lifespan.rst [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+are removed faster than new ones are being sent, due to the lifespan QoS, while the red square publisher history samples
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
 were removed by the Qos, and are no longer available to be sent to the subscriber, while the red square publisher samples
```

File: docs/lifespan.rst [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+are kept.
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
were kept in the history.
```

File: docs/lifespan.rst (hunk 16 above)
Comment (raquelalvarezbanos) [reply docs/lifespan.rst]: Did you change this finally?

File: docs/lifespan.rst [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+    - Set History to 100.
+    - Set Lifespan Duration to 50.
+
+Note that when a new subscriber matches with the publisher, due to the TRANSIENT_LOCAL durability, all the
+samples stored on the publisher history are sent automatically to the new subscriber.
+
+Now let's explain what is happening on Instance2 and Instance3. As you can see, the square subscriber history is
+filled rapidly, while the triangle subscriber one is filled at the same speed as the orange triangle publisher
+sends new samples. This is because in the second case, samples in the orange triangle publisher history were removed
+by the Qos, and are no longer available to sent to the subscriber, while the red square publisher history samples were
```
Comment (raquelalvarezbanos) [inline docs/lifespan.rst]:
```suggestion
by the Qos, and are no longer available to be sent to the subscriber, while the red square publisher history samples were
```
